src/platform/plugins/shared/es_ui_shared/public/request/use_request.test.helpers.tsx

```diff
@@ -52,28 +51,35 @@ const errorWithBodyResponse = { body: errorValue };
 export const createUseRequestHelpers = (): UseRequestHelpers => {
   // The behavior we're testing involves state changes over time, so we need finer control over
   // timing.
-  jest.useFakeTimers({ legacyFakeTimers: true });
-
-  const flushPromiseJobQueue = async () => {
-    // See https://stackoverflow.com/questions/52177631/jest-timer-and-promise-dont-work-well-settimeout-and-async-function
-    await Promise.resolve();
-  };
+  jest.useFakeTimers();
 
   const completeRequest = async () => {
     await act(async () => {
-      jest.runAllTimers();
-      await flushPromiseJobQueue();
+      await jest.runAllTimersAsync();
     });
   };
 
   const advanceTime = async (ms: number) => {
-    await act(async () => {
-      jest.advanceTimersByTime(ms);
-      await flushPromiseJobQueue();
-    });
+    // Split advancement into two separate act() phases so that effects scheduled
+    // by the first phase (e.g. scheduling the next poll) have a chance to run
+    // before we advance the second phase.
+    const first = Math.floor(ms / 2);
```

Comment:
I might be misunderstanding, but I think splitting the timer adv in two phases doesn't guarantee effects will fully settle because it only advances timers and ignores other things like promises created inside them. 

I wonder if perhaps something like:

```typescript
export const advanceTime = async (ms: number): Promise<void> => {
  await act(async () => {
    await jest.advanceTimersByTimeAsync(ms);
  });
  
  // flushhhhh
  await act(async () => {
    await Promise.resolve();
  });
};
```

Reply:
Done